Disable Bloom filter optimization by default

IagoAbal opened the pull request:
It conflicts with const-prop and sym-prop and appears not to provide a major net benefit nowadays.

To close #4670 we still need to tackle filter-irrelevant-rules interaction with const-prop.

Helps #4670
Helps PR #6179
Closes PA-1927

test plan:
make test # added one test

PR checklist:

- [x] Purpose of the code is [evident to future readers](https://semgrep.dev/docs/contributing/contributing-code/#explaining-code)
- [x] Tests included or PR comment includes a reproducible test plan
- [x] Documentation is up-to-date
- [x] A changelog entry was [added to changelog.d](https://semgrep.dev/docs/contributing/contributing-code/#adding-a-changelog-entry) for any user-facing change
- [x] Change has no security implications (otherwise, ping security team)

If you're unsure on any of this, please see:

- [Contribution guidelines](https://semgrep.dev/docs/contributing/contributing-code)!
- [One of the more specific guides located here](https://semgrep.dev/docs/contributing/contributing/)

emjin reviewed (approved):
Yay! Let's review in a month and if nobody has complained, delete it altogether.

IagoAbal committed:
Disable Bloom filter optimization by default

It conflicts with const-prop and sym-prop and appears not to provide a
net major benefit nowadays.

To close #4670 we still need to tackle filter-irrelevant-rules
interaction with const-prop.

Helps #4670
Helps PR #6179
Closes PA-1927

test plan:
make test # added one test

aryx commented on semgrep-core/src/core/Flag_semgrep.ml:
```diff
@@ -24,8 +24,30 @@ let filter_irrelevant_patterns = ref false
+ *   average of 9.5% faster, with the largest effect seen on mypy (15% faster).
+ * - In the remaining case (flask) there was no meaningful difference.
+ *
+ * That said take these %'s with a grain of salt, since we only did one run each,
```
great comment! and nice test.
